pkg/prometheus/agent/operator.go

```diff
@@ -282,6 +283,20 @@ func New(ctx context.Context, restConfig *rest.Config, c operator.Config, logger
 		return nil, fmt.Errorf("error creating statefulset informers: %w", err)
 	}
 
+	o.dsetInfs, err = informers.NewInformersForResource(
```

Comment:
we need to check if the feature gate is enabled or not

```suggestion
	if c.Gates.Enabled(PrometheusAgentDaemonSetFeature) {
		o.dsetInfs, err = informers.NewInformersForResource(
```